### PR TITLE
Allow different match identifiers

### PR DESCRIPTION
**Pull request by kelseyq**
adding an option to allow Twitter match identifiers

### CODE REVIEW

File: callisto/delivery/matching_validators.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        'who_to_follow',
+        'settings',
+        'search',
+        'tos',
+        'privacy',
+        'about',
+    ]
+
+
+def twitter_validation_function(value):
```
Comment (lynnco) [inline callisto/delivery/matching_validators.py]: I'm a bit concerned about the long term maintenance of keeping social url validators up to date? ie twitter changing its profile urls, and us not catching it for a few months

Comment (lynnco) [inline callisto/delivery/matching_validators.py]: or not changing, but adding an alias

Comment (lynnco) [inline callisto/delivery/matching_validators.py]: I'll add these thoughts to https://github.com/SexualHealthInnovations/callisto-core/issues/181, actually

Comment (kelseyq) [inline callisto/delivery/matching_validators.py]: sure, but this problem predates this PR (exists with facebook as well.) any ideas for mitigating?

Comment (lynnco) [inline callisto/delivery/matching_validators.py]: Yea this issue doesn't block this PR at all. I'll add my thoughts to that issue

Comment (lynnco) [inline callisto/delivery/matching_validators.py]: Yea that problem doesn't block this PR at all. I'll add my thoughts to #181 in a bit

File: callisto/delivery/forms.py [hunk 2]
```diff
@@ -160,59 +160,58 @@ def __init__(self, user, report, *args, **kwargs):
...
+        return ' or '.join([all_but_last, last])
 
 
 class SubmitToMatchingForm(forms.Form):
-    perp_name = forms.CharField(label="Perpetrator's Name",
-                                required=False,
+
+    def __init__(self, *args, **kwargs):
+        super(SubmitToMatchingForm, self).__init__(*args, **kwargs)
+        self.identifier_domain_info = getattr(settings, 'IDENTIFIER_DOMAINS', matching_validators.facebook_only)
```
Comment (lynnco) [inline callisto/delivery/forms.py]: just in case, I'd append `CALLISTO_` to `IDENTIFIER_DOMAINS`

Comment (lynnco) [inline callisto/delivery/forms.py]: also it occurs to me that we don't document our settings! I'll add that

**Pull request merged**